powerlevel9k.zsh-theme

```diff
@@ -973,18 +973,31 @@ build_right_prompt() {
     # Check if it is a custom command, otherwise interpet it as
     # a prompt.
     if [[ $element[0,7] =~ "custom_" ]]; then
-      "prompt_custom" "right" "$index" $element[8,-1]
+      "prompt_custom" "right" "$index" $element[8,-1] >> $POWERLEVEL9K_SOCKET
     else
-      "prompt_$element" "right" "$index"
+      "prompt_$element" "right" "$index" >> $POWERLEVEL9K_SOCKET
     fi
 
     index=$((index + 1))
+    if [[ "$POWERLEVEL9K_RPROMPT_RENDER_ASAP" == true ]]; then
+      kill -s USR1 $$
+    fi
   done
+  if [[ "$POWERLEVEL9K_RPROMPT_RENDER_ASAP" != true ]]; then
+    kill -s USR1 $$
+  fi
 }
 
+if [[ "$POWERLEVEL9K_DISABLE_RPROMPT" != true ]]; then
+  POWERLEVEL9K_SOCKET=$(mktemp)
```

Comment:
May I suggest a small improvement to the `mktemp` call?

It would be (a) nice to use a prefix for the temporary file to indicate where it comes from, e.g. `mktemp -t powerlevel_rprompt_$$`, and (b) check if `mktemp` succeeded and if it failed, make it fail silently, instead of blowing up the prompt e.g. `POWERLEVEL9K_SOCKET="$(mktemp -q -t powerlevel9k_rprompt_$$)"`

Overall, this block would like this:
```zsh
if [[ "$POWERLEVEL9K_DISABLE_RPROMPT" != true ]]; then
  POWERLEVEL_SOCKET="$(mktemp -q -t powerlevel_rprompt_$$)"
  if [ $? -eq 0 ]; then
    powerlevel9k_async() {
      : >! $POWERLEVEL_SOCKET
      build_right_prompt
    }
  else
    POWERLEVEL9K_RPROMPT_RENDER_ASAP=false
  fi
fi
```
I am sorry I couldn't test my suggestion. I hope you will give it some thought.